miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,590 @@
+
+    @property
+    def volume(self) -> int:
+        return self.data["volume"]
+
+    @property
+    def mute(self) -> bool:
+        return self.data["mute"]
+
+
+class RoidmiVacuumMiot(MiotDevice):
+    """Interface for Vacuum Eve Plus (roidmi.vacuum.v60)"""
+
+    mapping = _MAPPING
+
+    @command(
+        default_output=format_output(
+            "\n",
+            "Battery level: {result.battery_level}\n"
+            "Brush life level: {result.brush_life_level}\n"
+            "Brush left time: {result.brush_left_time}\n"
+            "Charging state: {result.charging_state.name}\n"
+            "Device fault: {result.device_fault.name}\n"
+            "Device status: {result.device_status.name}\n"
+            "Filter left level: {result.filter_left_time}\n"
+            "Filter life level: {result.filter_life_level}\n"
+            "Operating mode: {result.sweep_mode.name}\n"
+            "Right side cleaning brush left time: {result.brush_left_time2}\n"
+            "Right side cleaning brush life level: {result.brush_life_level2}\n"
+            "Left side cleaning brush left time: {result.brush_left_time3}\n"
+            "Left side cleaning brush life level: {result.brush_life_level3}\n"
+            "Cleaning mode: {result.cleaning_mode.name}\n"
+            "Sweep type: {result.sweep_type.name}\n"
+            "Path mode: {result.path_mode.name}\n"
+            "Sweep mode: {result.sweep_mode.name}\n"
+            "Mop present: {result.mop_present}\n"
+            "work_station_freq: {result.work_station_freq}\n"
+            "timing: {result.timing}\n"
+            "clean_area: {result.clean_area}\n"
+            # "uid: {result.uid}\n"
+            "auto_boost: {result.auto_boost}\n"
+            "forbid_mode: {result.forbid_mode}\n"
+            "Water level: {result.water_level.name}\n"
+            # "Unknown siid8_13 [sec]: {result.siid8_13}\n"
+            # "Unknown siid8_14 [uint32]: {result.siid8_14}\n"
+            "clean_counts: {result.clean_counts}\n"
+            # "Unknown siid8_19 [sec]: {result.siid8_19}\n"
+            "Double clean: {result.double_clean}\n"
+            "Edge sweep: {result.edge_sweep}\n"
+            "Led switch: {result.led_switch}\n"
+            "Lidar collision: {result.lidar_collision}\n"
+            "Station key: {result.station_key}\n"
+            "Station led: {result.station_led}\n"
+            "Current audio: {result.current_audio}\n"
+            "Progress: {result.progress}\n"
+            # "Voice config: {result.voice_conf}\n"
+            # "Switch status: {result.switch_status.name}\n"
+            "Volume: {result.volume}\n" "Mute: {result.mute}\n",
+        )
+    )
+    def status(self) -> RoidmiVacuumStatus:
+        """State of the vacuum."""
+
+        return RoidmiVacuumStatus(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                # max_properties limmit to 10 to avoid "Checksum error" messages from the device.
```

Comment:
This might be fixed by the recent #1075 - could you please confirm if that's the case?

Reply:
For Roidmi Eve there are two reasons for getting "checksum error": 1) is too many properties at a time (here I use max_properties), 2) Too many properties in total.
Commit #1075 did not affect No. 1, so I need to keep max_properties, but it did solved No. 2 (This means that I can uncomment all properties in MiotMapping). Thank you very much for telling me about this fix.